src/Data/Nat/Bounded.agda

```diff
@@ -0,0 +1,81 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Bounded natural numbers, and their equivalence to `Fin`
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+module Data.Nat.Bounded where
```

Comment:
So to touch on your question of whether this belongs in the library or a readme file, I guess my question is what circumstances is this easier to work with than `Fin`? In general we try to avoid duplicate but equivalent definitions unless there is clear reason that sometimes one representation is easier to work with than the other...

Reply:
Thanks, Matthew! I can think of two or three, some already in the initial preamble (which was really about helping someone solve a related problem of representation, where `Fin` didn't seem to fit well):
* fixed-width arithmetic (or even: its extension to variable-widths, but in a way which retains predictable/computable bounds);
* `Digit` bases for arithmetic, as a specialised instance/mode of use of the above;
* revising the 'specification' of `Data.Nat.DivMod`, to avoid the marshalling/unmarshalling required by the use of `Fin` to ensure that `mod` is suitably bounded.

Now, none of these could be said to be a deal-breaker, but I increasingly wonder whether `Fin` is better suited to its use as an *indexing* type (for `Vec`, `List`, de Bruijn representations, `Reflection`/deeply embedded syntaxes more generally, etc.), with `ℕ<` as an *arithmetic* type/`Number` instance...

Perhaps other pros (and cons!) might be lurking off-stage...

Reply:
Previously...

> ... UPDATED:
> 
>     * re: 3rd point above. This might (eventually) lead to the deprecation of `Data.Nat.DivMod.WithK` (no longer any need to consider erasing proofs of `_≤″_`...?) and hence perhaps to the deprecation of `_≤″_` _altogether_, cf. [Remove all external use of `_≤″_` beyond `Data.Nat.*` #2262](https://github.com/agda/agda-stdlib/pull/2262)

This cashes out as the following `breaking` (because type-signature-changing... hence not committed to this PR) simplifications to `Data.Nat.DivMod`  and `Data.Nat.DivMod.WithK`:
```agda
record DivMod (dividend divisor : ℕ) : Set where
  constructor result
  field
    quotient  : ℕ
    remainder : ℕ< divisor
    property  : dividend ≡ ⟦ remainder ⟧ + quotient * divisor

  nonZeroDivisor : NonZero divisor
  nonZeroDivisor = nonZeroIndex remainder


infixl 7 _div_ _mod_ _divMod_

_div_ : (dividend divisor : ℕ) .{{_ : NonZero divisor}} → ℕ
_div_ = _/_

_mod_ : (dividend divisor : ℕ) .{{_ : NonZero divisor}} → ℕ< divisor
m mod n = ⟦ m % n ⟧< (m%n<n m n)

_divMod_ : (dividend divisor : ℕ) .{{_ : NonZero divisor}} →
           DivMod dividend divisor
m divMod n = result (m / n) (m mod n) $ m≡m%n+[m/n]*n m n
```
and (`WithK`)
```agda
_divMod_ : (dividend divisor : ℕ) → .{{ NonZero divisor }} →
           DivMod dividend divisor
m divMod n = result (m / n) (m mod n) $ ≡-erase $ m≡m%n+[m/n]*n m n
```

Suggestions on how to proceed, if, like me, you find these compelling.